Release notes v0.1.6

Anish9901 opened the pull request:
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, open one before creating this pull request. -->
Release notes for v0.1.6

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

Anish9901 commented:
I forgot to mention this in our last team meeting, do we need to add another page in our docs for v0.1.6 upgrade instructions if they are the same as v0.1.5?

pavish commented:
> I forgot to mention this in our last team meeting, do we need to add another page in our docs for v0.1.6 upgrade instructions if they are the same as v0.1.5?

@Anish9901 

They're not the same as v0.1.5. There are changes in steps involved for Install from scratch. It's tracked in this issue: https://github.com/mathesar-foundation/mathesar/issues/3505

kgodey commented:
We should also add a Basecamp task to 0.1.7 and the template to make sure we have upgrade instructions for the release, @Anish9901 can you do this?

kgodey commented:
Some thoughts based on a quick skim, I'll review in more detail next week once these are addressed:
- In the release summary, I think we should also highlight Postgres 16 support and also making longer text easier to work with via the record page.
- Please make sure to capitalize words like "Django", "Docker", and "Japanese". You may want to do a general grammar check too.

Anish9901 commented:
@kgodey I've addressed your concerns in b5fc1a4 & f56a96a, The CI is failing as the linked page for upgrading to 0.1.6 doesn't exist yet and will be added once #3507 is merged. Please feel free to go through another round of review.

mathemancer commented on docs/docs/releases/0.1.6.md:
```diff
@@ -0,0 +1,36 @@
+
+- Mathesar is now compatible with Python versions: 3.10 and 3.11 along with 3.9 _[#3478](https://github.com/mathesar-foundation/mathesar/pull/3478)_
+- Added support for Japanese translations _[#3486](https://github.com/mathesar-foundation/mathesar/pull/3486)_, _[#3484](https://github.com/mathesar-foundation/mathesar/pull/3484)_, _[#3483](https://github.com/mathesar-foundation/mathesar/pull/3483)_, _[#3472](https://github.com/mathesar-foundation/mathesar/pull/3472)_
+- NodeJS is no longer a requirement for building Mathesar from source _[#3474](https://github.com/mathesar-foundation/mathesar/pull/3474)_
```
This isn't actually what the linked PR is about is it? It's about removing Node from the prod Docker image.

mathemancer commented on docs/docs/releases/0.1.6.md:
```diff
@@ -0,0 +1,36 @@
+
+## Bug fixes
+
+- Fixed regression within the record selector where records won’t get filtered based on input _[#3488](https://github.com/mathesar-foundation/mathesar/pull/3488)_
```
Quite long, wraps on most screens. Try

```suggestion
- Fixed record selector filtering bug _[#3488](https://github.com/mathesar-foundation/mathesar/pull/3488)_
```

mathemancer commented on docs/docs/releases/0.1.6.md:
```diff
@@ -0,0 +1,36 @@
+
+- Fixed regression within the record selector where records won’t get filtered based on input _[#3488](https://github.com/mathesar-foundation/mathesar/pull/3488)_
+- Fixed failures while adding a connection along with sample schema(s), whose name(s) would collide with schema(s) already present in the database _[#3490](https://github.com/mathesar-foundation/mathesar/pull/3490)_
+- Fixed record selector's text inputs from growing taller _[#3495](https://github.com/mathesar-foundation/mathesar/pull/3495)_
```
Incorrect grammar. @seancolsen Do you have a preferred rewording for this one?

mathemancer commented on docs/docs/releases/0.1.6.md:
```diff
@@ -0,0 +1,36 @@
+- Added MkDocs edit URI _[#3482](https://github.com/mathesar-foundation/mathesar/pull/3482)_
+- Bumped Django from 4.2.8 to 4.2.10 _[#3492](https://github.com/mathesar-foundation/mathesar/pull/3492)_
+- Removed stale code coverage badge _[#3491](https://github.com/mathesar-foundation/mathesar/pull/3491)_
+- Added GitHub workflow to attach pre-built static files with new releases _[#3489](https://github.com/mathesar-foundation/mathesar/pull/3489)_ 
```
This is actually the part (along with docs) that removes the NodeJS dependency for building from source.

mathemancer reviewed (changes requested):
Format-wise, it's fine.

The only blocking change I'd request is to sort out the links for removing the NodeJS dependency when building from source.

Generally, I think it's worth the effort to try to make the descriptions of PRs short and dense enough that they don't need to wrap except in edge cases.

mathemancer reviewed (approved):
Okay, this is good to go from my end. Thank you!

kgodey reviewed (commented):
I made some changes to the summary, this looks okay with me, but I haven't done a line-by-line review of each of the changes listed. 

I will leave it to @seancolsen to do the line-by-line review since he wrote the previous release notes, and judging by https://github.com/mathesar-foundation/mathesar-wiki/commit/788361481eb3b937b64b54da2391ad46734fc5f1 he has some opinions about this.

kgodey commented:
> We should also add a Basecamp task to 0.1.7 and the template to make sure we have upgrade instructions for the release, @Anish9901 can you do this?

Also @Anish9901 I didn't see a response to this so I just did it myself.

Anish9901 commented:
> > We should also add a Basecamp task to 0.1.7 and the template to make sure we have upgrade instructions for the release, @Anish9901 can you do this?
> 
> Also @Anish9901 I didn't see a response to this so I just did it myself.

I added it under "Documentation" @kgodey 
<img width="546" alt="Screenshot 2024-03-27 at 2 49 52 AM" src="https://github.com/mathesar-foundation/mathesar/assets/52523023/815f87d4-2908-476b-9211-f7193bba994f">

seancolsen reviewed (commented):
I have a number of small points of critique. These are mostly small, detail-level things.

As much as this review is a critique of your work in writing the release notes, it's also a critique of the release notes _process_, i.e. the [Writing Release Notes](https://wiki.mathesar.org/release/notes) wiki page that I created. I'd like to find ways to improve (or better document) this process where we can. For that reason, I'm dwelling on this PR review a bit more than I normally would. Hypothetically, I could quickly push a commit to this PR implementing my suggested changes and call it a day. But I want to get our process honed so that it's easy to hand off. Towards that end, today I added a section to that wiki page which lists [guidelines](https://wiki.mathesar.org/release/notes/#guidelines) to follow when writing the release notes. I'll admit that even I didn't even follow all these guidelines when authoring the 0.1.5 release notes — which is one reason I think it'll be helpful to keep in the document as a sort of checklist. For the sake of speed (and because the stakes here are quite low), I'll assume that nobody on the team wishes to contest any of these new release notes guidelines, but I'm happy to discuss as needed. 

My critique:

- I'm seeing signs that perhaps you didn't follow the first step in the process. Did you run `./find_missing_prs.sh 0.1.6`?

    - If you _had_ run that command, then I would expect to see double-quoted strings after all PR URLs, as shown in the 0.1.5 release notes. This is special markdown which generates tool tips like this:

        ![image](https://github.com/mathesar-foundation/mathesar/assets/42411/4425d403-8cfd-4c2d-8a4a-2a651d679f93)

        Those tool tips are there to make it a little bit easier for reader to see the original PR titles.

    - Also, there appear to be some PRs missing from the release notes: #3463 #3513 #3503 #3469 #3476. I see those PRs when I run the script in this PR's branch.

    If you _did_ run the script, then I'd like to troubleshoot why it's not giving the behavior I would expect. If you didn't run the script, I'd like to understand why. Do you have an opinion against this script-based approach? Or could we improve our release notes process documentation to make this step more obvious?

- I suggest modifying the order of items within the Improvements section as follows:

    - i18n
    - auto-expanding text fields
    - Python versions
    - NodeJS

    This order focuses on user-facing changes first and better showcases our recent work on i18n, which I think readers are likely to find interesting and impressive.

- Adjust the title of the i18n item to improve clarity and become grammatically parallel with the other list items:

    > You can now configure Mathesar's UI to display in Japanese

- Add descriptions for all four items in the improvements section. For example, here's a proposed description for the i18 improvement:

    > The language setting is stored per-user and can be modified when logging in or when editing a user. This changes the text displayed on buttons and other UI elements within Mathesar. It does _not_ change the display of data within your database (e.g. table names, column names, and cell values). We are hoping to support more languages beyond English and Japanese eventually. Please reach out to us if your are interested in helping to add more translations!

- Add screenshots for the "auto-expanding text fields" and "i18n" features. I sent a screenshot [here](https://matrix.to/#/!kpLRqiyQwsndKhBIvd:matrix.mathesar.org/$d8k61LujkYz2Brm-SzPzybGmxWVB4nXhLN8PE098KW8?via=matrix.mathesar.org&via=matrix.org) for the i18n feature that you can use.

- Re-categorize mid-cycle regressions.

    - Remove entries for #3488 and #3495. Instead, list these within the "auto-expanding text fields" improvement. 
    - Remove the entry for #3501. Instead, list it under the "i18n" improvement.

    See my (newly written) guideline for more rationale about these changes.

kgodey commented:
I think your guidelines make sense, @seancolsen. Since there's not much left to do in the release process, I think it's fine to make these changes before release, unless @Anish9901 thinks they will take more than a day.

Anish9901 commented:
> If you didn't run the script, I'd like to understand why. Do you have an opinion against this script-based approach? Or could we improve our release notes process documentation to make this step more obvious?

I wasn't able to get the script running on a Mac @seancolsen, That's why I decided not to use it, I wasn't aware that it added a bunch of metadata about linked issues. I've got it to work using a linux machine now.

Here's the specific error that I was facing on a Mac:
```
anish@appy releases % ./find_missing_prs.sh 0.1.6
sed: 1: "0.1.6.md": invalid command code .
date: illegal option -- d
usage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]
            [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
            [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]
grep: invalid option -- P
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
Binder Error: Table "pr" does not have a column named "mergeCommit"
LINE 7:     ON commit.hash = pr.mergeCommit
                             ^
No missing PRs
```

seancolsen commented:
@Anish9901 it looks like you found some portability issues with my bash script. Wonderful! Instead of ignoring this script, I'd like to fix it. The hover text on PR hyperlinks is cool, but the main value proposition of the script is that it makes it _really fast_ to generate the release notes. I've opened #3515 to track fixes to the script. I don't imagine this being too hard, so I think we could take it up for the next release.

In the mean time, it's great that you've found a work-around using Linux. I assume you can use this workflow to push changes that address my critique. Let me know if you need any help.

seancolsen reviewed (approved):
Splendid! @Anish9901 you've addressed all of my feedback perfectly.

I merged `0.1.6` into this branch which I think should fix the CI errors. (Mkdocs was complaining when trying to build in strict mode because the upgrade instructions page was a dead link since that page got implemented separately as part of #3507.) Assuming those CI errors resolve, then this is ready to merge.
